pkg/ingress/admission/api/types.go

```diff
@@ -0,0 +1,19 @@
+package api
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// IngressAdmissionConfig is the configuration for the the ingress
+// controller limiter plugin. It changes the behavior of ingress
+//objects to behave better with openshift routes and routers.
+//*NOTE* Disabling this plugin causes ingress objects to behave
+//the same as in upstream kubernetes
+type IngressAdmissionConfig struct {
+	unversioned.TypeMeta
+
+	//UpstreamHostnameUpdate when true causes updates that attempt
+	//to add or modify hostnames to succeed. Otherwise those updates
+	//fail in order to ensure hostname behavior
```

Comment:
This needs to be clarified to explain that we don't allow addition or renaming of hostnames in the ingress object when set to false.  If set to true then we allow the changes, but that has security implications.

What is the default?